projects/vdk-plugins/vdk-impala/src/vdk/plugin/impala/impala_plugin.py

```diff
@@ -1,161 +1,94 @@
-            "list of auth cookie names used by Impala and Hive. If 'impala_auth_cookie_names' is explicitly set to "
-            "an empty value (None, [], or ''), VDK won't attempt to do cookie based authentication. "
-            "Currently cookie-based authentication is only supported for GSSAPI over http."
-        ),
-    )
-    config_builder.add(
-        key="IMPALA_RETRIES",
-        default_value=3,
-        description="The number of retries VDK will attempt when connecting to or running a query against Impala.",
+        host=configuration.host(),
+        port=configuration.port(),
+        database=configuration.database(),
+        timeout=configuration.timeout(),
+        use_ssl=configuration.use_ssl(),
+        ca_cert=configuration.ca_cert(),
+        auth_mechanism=configuration.auth_mechanism(),
+        user=configuration.user(),
+        password=configuration.password(),
+        kerberos_service_name=configuration.kerberos_service_name(),
+        krb_host=configuration.krb_host(),
+        use_http_transport=configuration.use_http_transport(),
+        http_path=configuration.http_path(),
+        auth_cookie_names=configuration.auth_cookie_names(),
+        retries=configuration.retries(),
     )
 
 
 @click.command(name="impala-query", help="executes SQL query against Impala")
 @click.option("-q", "--query", type=click.STRING, required=True)
 @click.pass_context
 def impala_query(ctx: click.Context, query):
-    click.echo(
-        tabulate(
-            _connection_by_configuration(ctx.obj.configuration).execute_query(query)
-        )
-    )
+    impala_cfg = ImpalaPluginConfiguration(ctx.obj.configuration)
+    click.echo(tabulate(_connection_by_configuration(impala_cfg).execute_query(query)))
 
 
-@hookimpl
-def vdk_command_line(root_command: click.Group):
-    root_command.add_command(impala_query)
+class ImpalaPlugin:
+    @hookimpl
+    def vdk_configure(self, config_builder: ConfigurationBuilder) -> None:
+        add_definitions(config_builder)
 
+    @hookimpl
+    def vdk_command_line(self, root_command: click.Group):
+        root_command.add_command(impala_query)
 
-@hookimpl
-def initialize_job(context: JobContext) -> None:
-    context.connections.add_open_connection_factory_method(
-        "IMPALA",
-        lambda: _connection_by_configuration(context.core_context.configuration),
-    )
+    @hookimpl
+    def initialize_job(self, context: JobContext) -> None:
+        self._impala_cfg = ImpalaPluginConfiguration(
+            context.core_context.configuration
+        )  # necessary for the query decoration hook
+        context.connections.add_open_connection_factory_method(
+            "IMPALA",
+            lambda: _connection_by_configuration(context.core_context.configuration),
+        )
 
+    @hookimpl
+    def db_connection_recover_operation(self, recovery_cursor: RecoveryCursor) -> None:
```

Comment:
```suggestion
    @hookimpl
    @staticmethod
    def db_connection_recover_operation(recovery_cursor: RecoveryCursor) -> None:
```

nit: you may consider adding `@staticmethod` annotation to get rid of the codacy warning